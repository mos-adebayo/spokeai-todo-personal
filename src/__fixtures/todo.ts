export const actionItems = [
  {
    isDone: true,
    description: "description 1"
  },
  {
    isDone: false,
    description: "description 2"
  }
];

export const todo = {
  id: "1234567890",
  title: "Title",
  items: actionItems
};
