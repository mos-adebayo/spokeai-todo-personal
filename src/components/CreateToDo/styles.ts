import styled from "styled-components";

export const FormWrapper = styled.div`
  width: 50%;
  margin: 20px auto;
  
  .form-control:focus {
    box-shadow: none;
  }
`;
