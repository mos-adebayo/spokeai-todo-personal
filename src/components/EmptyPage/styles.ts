import styled from "styled-components";

export const Wrapper = styled.div`
  width: 100%;
  margin: 20px auto;
  text-align: center;

  @media screen and (min-width: 768px) {
    width: 80%;
  }

  @media screen and (min-width: 992px) {
    width: 60%;
  }

  @media screen and (min-width: 1200px) {
    width: 50%;
  }
`;

export const LinkText = styled.span`
  cursor: pointer;
  text-decoration: underline;
`;
