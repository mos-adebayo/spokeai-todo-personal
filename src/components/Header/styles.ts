import styled from "styled-components";
import { gradientPrimaryBackground } from "../../util/stylesheet";

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  padding: 45px 20px;
  height: 100px;
  margin-bottom: 25px;
  font-family: "Mochiy Pop One", sans-serif;

  ${gradientPrimaryBackground};
`;
