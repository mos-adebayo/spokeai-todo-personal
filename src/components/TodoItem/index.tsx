import React, { useState, useEffect } from "react";
import {
  CheckBox,
  HeaderWrapper,
  ItemWrapper,
  Text,
  Title,
  Wrapper
} from "./styles";
import { ProgressBar } from "react-bootstrap";
import { getProgressStatus } from "../../util/helper";

type Props = {
  todo: TaskItemType;
};

const TodoItem: React.FC<Props> = ({ todo }) => {
  const [progress, setProgress] = useState(0);
  const [status, setStatus] = useState("danger");

  useEffect(() => {
    const completedItems = todo.items.filter((item) => item.isDone).length;
    const progressPercentage =
      completedItems === 0
        ? 0
        : Math.ceil((completedItems / todo.items.length) * 100);
    setProgress(progressPercentage);
  }, [todo]);

  useEffect(() => {
    setStatus(getProgressStatus(progress));
  }, [progress]);

  return (
    <Wrapper>
      <HeaderWrapper>
        <Title>{todo.title}</Title>
        {todo.items.length > 0 && (
          <ProgressBar
            striped
            now={progress}
            variant={status}
            label={`${progress}%`}
          />
        )}
      </HeaderWrapper>

      {todo.items.map((item, key) => (
        <ItemWrapper key={key}>
          <CheckBox checked={item.isDone} />
          <Text>{item.description}</Text>
        </ItemWrapper>
      ))}
    </Wrapper>
  );
};

export default TodoItem;
