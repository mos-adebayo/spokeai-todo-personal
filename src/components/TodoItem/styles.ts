import styled from "styled-components";

export const Wrapper = styled.div`
  background: #fff;
  border-radius: 12px;
  box-shadow: rgb(166 171 189 / 20%) 0 10px 20px;
  padding: 15px 30px;
  margin-bottom: 15px;
`;

export const Title = styled.h4`
color: #49486c;
`;

export const Text = styled.p`
color: #49486c
`;
