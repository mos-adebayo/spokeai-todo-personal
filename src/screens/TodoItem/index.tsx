import React from 'react';

const TodoItem = () => {
    return (
        <div>
            <h2>TODO Title</h2>
            <p>
                Todo description.
            </p>
        </div>
    );
};

export default TodoItem;
