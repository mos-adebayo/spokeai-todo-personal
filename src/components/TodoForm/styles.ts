import styled, { css } from "styled-components";
import { Button } from "react-bootstrap";
import { CheckBox } from "../TodoItem/styles";

export const AddButton = styled(Button)`
  &:focus {
    box-shadow: none;
  }

  background: linear-gradient(
    77.07deg,
    rgb(69, 183, 204) 0%,
    rgb(75, 207, 187) 100%
  );
  outline: none;
  border: none;
`;

export const AddButtonWrapper = styled.div`
  position: absolute;
  right: 50px;
  bottom: 50px;

  display: flex;
  justify-content: center;
  align-items: center;

  color: #ffffff;
  cursor: pointer;

  width: 100px;
  height: 100px;
  font-size: 80px;
  border-radius: 50%;
  background: linear-gradient(
    77.07deg,
    rgb(69, 183, 204) 0%,
    rgb(75, 207, 187) 100%
  );
  box-shadow: rgb(52 58 64 / 40%) 3px 1px 13px 0;

  @media screen and (max-width: 767px) {
    width: 80px;
    height: 80px;
    font-size: 60px;
  }
`;

export const FormWrapper = styled.div`
  margin: 20px 0;

  .form-control:focus {
    box-shadow: none;
  }
`;

export const ItemWrapper = styled.div`
  border-bottom: 1px solid #dfdfdf;
  margin: 0 -16px;
  padding: 10px 16px 8px;

  ${CheckBox} {
    margin-right: 5px;
  }

  &:first-of-type {
    border-top: 1px solid #dfdfdf;
  }
`;

const editItemWrapper = css`
  ${ItemWrapper} {
    margin: 0;
  }
`;

export const ItemsWrapper = styled.div<{ editMode?: boolean }>`
  margin-top: 15px;

  .form-control {
    height: 1.5rem;
    border: none;
  }

  ${({ editMode }) => editMode && editItemWrapper};
`;
