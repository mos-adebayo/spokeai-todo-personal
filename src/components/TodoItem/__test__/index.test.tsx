import React from "react";
import {shallow, ShallowWrapper} from "enzyme";
import TodoItem from "../index";
import {todo} from "../../../__fixtures/todo";
import {ItemWrapper} from "../styles";


describe("<TodoItem />", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
        <TodoItem todo={todo}/>
    );
  })

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render action items correctly", () => {
    const items = wrapper.find(ItemWrapper);

    expect(items.length).toEqual(2);
  });
});
