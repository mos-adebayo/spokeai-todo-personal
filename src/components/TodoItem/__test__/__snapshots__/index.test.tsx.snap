// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem /> should match snapshot 1`] = `
<styled.div
  completed={false}
>
  <styled.div>
    <styled.h4>
      Title
    </styled.h4>
    <ProgressBar
      animated={false}
      isChild={false}
      label="0%"
      max={100}
      min={0}
      now={0}
      striped={true}
      variant="danger"
      visuallyHidden={false}
    />
  </styled.div>
  <styled.div
    checked={true}
    key="0"
  >
    <styled.span
      checked={true}
    />
    <styled.span>
      description 1
    </styled.span>
  </styled.div>
  <styled.div
    checked={false}
    key="1"
  >
    <styled.span
      checked={false}
    />
    <styled.span>
      description 2
    </styled.span>
  </styled.div>
</styled.div>
`;
